projects/core/src/lib/operators/dialogs/helpers/multi-layer-selection/multi-layer-selection.component.html

```diff
@@ -10,19 +10,31 @@ <h4>Select input {{ title }}</h4>
     </div>
 </div>
 <div *ngIf="hasLayers | async; else noInputAvailable" fxLayout="column">
-    <mat-form-field *ngFor="let number of [].constructor(this.selectedLayers.value.length); let i = index">
-        <mat-label>{{ inputNaming(i) }}</mat-label>
-        <mat-select
-            class="mat-select-{{ i }}"
-            [(value)]="selectedLayers.value[i]"
-            (selectionChange)="updateLayer(i, $event.value)"
-            (onBlur)="onBlur()"
-        >
-            <mat-option class="mat-option-{{ i }}" *ngFor="let layer of filteredLayers | async" [value]="layer">{{
-                layer.name
-            }}</mat-option>
-        </mat-select>
-    </mat-form-field>
+    <ng-container *ngFor="let number of [].constructor(this.selectedLayers.value.length); let i = index">
+        <div fxLayout="row">
+            <mat-form-field fxFlex>
+                <mat-label>{{ inputNaming(i) }}</mat-label>
+                <mat-select
+                    class="mat-select-{{ i }}"
+                    [(value)]="selectedLayers.value[i]"
+                    (selectionChange)="updateLayer(i, $event.value)"
+                    (onBlur)="onBlur()"
+                >
+                    <mat-option class="mat-option-{{ i }}" *ngFor="let layer of filteredLayers | async" [value]="layer">{{
+                        layer.name
+                    }}</mat-option>
+                </mat-select>
+            </mat-form-field>
+            <button type="button" mat-icon-button class="info" (click)="toggleExpand(i)">
+                <mat-icon *ngIf="!layerDetails[i].expanded">expand_more</mat-icon>
+                <mat-icon *ngIf="layerDetails[i].expanded">expand_less</mat-icon>
+            </button>
+        </div>
+        <geoengine-layer-collection-layer-details
```

Comment:
The expression now has only one input, so there cannot be any confusion about which band corresponds to which letter.

But I agree that it would be nice to show the individual bands too. But as there can be very many bands it's not directly clear to me how to do this. A popup? An iframe (something to scroll I mean)? A load more button?

Reply:
the expression has one input with multiple bands. How do you know the mapping of band to letter? The band number is not that helpful if it does not correspond to a satellite or dataset directly.
Something like a table with pagination maybe? https://material.angular.io/components/table/overview#table-sorting

Reply:
i created a new task